File: go/p007.go

package main
func Is_prime(n int64) bool {
	if n<=1{
		return false
	}
    if n==2{
    	return true;
    }
    if n%2==0 {
    	return false
    }
		var i int64=3
		for ;i*i<=n;i+=2{
			if n%i==0{
				return false
			}
		}
		return true
	

}
func compute() int64{
	var index int64=1
	count:=0
	for{
		index+=1
		if Is_prime(index){
			count+=1
			if(count==10001){
				return index
			}
		}
	}
}


func main(){

	println(compute())
}

File: cpp/p007.cc
/*  
* Solution to Project Euler problem 7
* by mathm   
*/

#include <iostream>


bool is_prime(long n){
	if (n<=1)  return false;
	if(n==2)  return true;
	if(n%2==0) return false;

	for(int i=3;i*i<=n;i+=2){
		if(n%i==0)  return false; 
	}

	return true;
}

long compute(){
	int index=1;
	int count=0;
	for(;;){
		index+=1;
		if(is_prime(index)){
			count++;
			if(count==10001)
				return index;
		}
	}
}


int main(){
	using std::cout;
	using std::endl;
	cout<<compute()<<endl;
}